src/utils.ts

```diff
@@ -341,50 +344,115 @@ const getAllPromptTemplates = async (
 };
 
 export const runWorkflowRequest = async ({
-   workflow_name,
-   input_variables,
-   metadata = {},
-   workflow_label_name = null,
-   workflow_version_number = null,
-   api_key,
- }: {
-  workflow_name: string;
-  input_variables: Record<string, any>;
-  metadata?: Record<string, string>;
-  workflow_label_name?: string | null;
-  workflow_version_number?: number | null;
-  api_key: string;
-}): Promise<WorkflowResponse> => {
+  workflow_name,
+  input_variables,
+  metadata = {},
+  workflow_label_name = null,
+  workflow_version_number = null,
+  return_all_outputs = false,
+  api_key,
+  timeout = 120000, // Default timeout is 2 minutes in milliseconds
+}: RunWorkflowRequestParams): Promise<WorkflowResponse> => {
   const payload = {
     input_variables,
     metadata,
     workflow_label_name,
     workflow_version_number,
+    return_all_outputs,
   };
 
+  const headers = {
+    'X-API-KEY': api_key,
+    'Content-Type': 'application/json',
+  }
+
   try {
-    const response = await fetch(`${URL_API_PROMPTLAYER}/workflows/${workflow_name}/run`, {
+    // Start the workflow by making a POST request
+    const response = await fetch(`${URL_API_PROMPTLAYER}/workflows/${encodeURIComponent(workflow_name)}/run`, {
       method: 'POST',
-      headers: {
-        'X-API-KEY': api_key,
-        'Content-Type': 'application/json',
-      },
+      headers: headers,
       body: JSON.stringify(payload),
     });
 
     if (response.status !== 201) {
       const errorData = await response.json().catch(() => ({}));
-      throw new Error(`Failed to run workflow: ${errorData.error || response.statusText}`);
+      return {success: false, message: "Failed to run workflow"}
```

Comment:
<img width="430" alt="Screenshot 2024-10-21 at 12 07 48 PM" src="https://github.com/user-attachments/assets/282dd7b9-624e-4241-b9e2-288b931086dc">

Reply:
Let's use the error message from the API when we can as it provides more details to the user on what went wrong.

<img width="1207" alt="Screenshot 2024-10-21 at 1 49 26 PM" src="https://github.com/user-attachments/assets/a4b6b4c5-9cc9-448f-b804-53498fef1eb3">